controllers/drpolicy_controller.go

```diff
@@ -77,7 +77,8 @@ func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, fmt.Errorf("finalizer add update: %w", u.validatedSetFalse("FinalizerAddFailed", err))
 		}
 
-		if reason, err := validate(ctx, drpolicy, r.APIReader, r.ObjectStoreGetter, req.NamespacedName.String()); err != nil {
+		reason, err := validateDRPolicy(ctx, drpolicy, r.APIReader, r.ObjectStoreGetter, req.NamespacedName.String())
```

Comment:
(nit) `err` gets defined here for the main function body, rest of the function has `err` in it's own sub-blocks, would like it if the same was retained here as well. (possibly `reason` in the future needs to be outside the block for metroDR and hence both get defined here)

Reply:
Yes, the reason is coming in from one of the called functions and we would like to keep it that way. I think we should refactor the other two calls also at some point in the future.